### PR TITLE
[EOC-551] Gmail sending email notifications implementation

### PR DESCRIPTION
**Pull request by alemikolo**
For this functionality to work, we have to instal `agenda`, `nodemailer`, `googleapis`, create Gmail account for the app and add save these fields to env file:
```
GOOGLE_CLIENT_ID=''
GOOGLE_CLIENT_SECRET=''
GOOGLE_REFRESH_TOKEN=''
GOOGLE_API_USER=''
HOST=''

```

### CODE REVIEW

File: server/mailer/index.js [hunk 1]
```diff
@@ -1,12 +1,51 @@
...
+  SENDGRID_API_KEY
+} = process.env;
+const {
+  auth: { OAuth2 }
+} = google;
+
+const oauth2Client = new OAuth2(
+  GOOGLE_CLIENT_ID,
+  GOOGLE_CLIENT_SECRET,
+  'https://developers.google.com/oauthplayground'
```
Comment (marekrozmus) [inline server/mailer/index.js]: is this correct?

File: server/mailer/index.js [hunk 2]
```diff
@@ -1,12 +1,47 @@
...
 const weeklyReportContent = require('./reports/weekly-content');
 const { FULL_PROJECT_NAME, PROJECT_NAME } = require('../common/variables');
 
-const { SENDGRID_API_KEY } = process.env;
+const {
+  GOOGLE_CLIENT_ID,
+  GOOGLE_CLIENT_SECRET,
+  GOOGLE_REFRESH_TOKEN,
+  GOOGLE_API_USER,
+  SENDGRID_API_KEY
```
Comment (adamklepacz) [inline server/mailer/index.js]: sort

File: server/mailer/index.js [hunk 3]
```diff
@@ -1,12 +1,47 @@
...
+} = process.env;
+const {
+  auth: { OAuth2 }
+} = google;
+
```
Comment (adamklepacz) [inline server/mailer/index.js]: remove thath empty line

File: server/mailer/index.js [hunk 4]
```diff
@@ -1,12 +1,47 @@
...
+const oauth2Client = new OAuth2(GOOGLE_CLIENT_ID, GOOGLE_CLIENT_SECRET, '/');
+
+const getMailer = () => {
+  oauth2Client.setCredentials({
+    refresh_token: GOOGLE_REFRESH_TOKEN
+  });
+
+  const accessToken = oauth2Client.getAccessToken();
+
```
Comment (adamklepacz) [inline server/mailer/index.js]: remove that empty line

File: server/mailer/index.js [hunk 5]
```diff
@@ -1,12 +1,47 @@
...
+  const transport = nodemailer.createTransport({
+    service: 'gmail',
+    auth: {
+      type: 'OAuth2',
+      user: GOOGLE_API_USER,
+      clientId: GOOGLE_CLIENT_ID,
+      clientSecret: GOOGLE_CLIENT_SECRET,
+      refreshToken: GOOGLE_REFRESH_TOKEN,
+      accessToken
```
Comment (adamklepacz) [inline server/mailer/index.js]: sort

File: server/mailer/index.js [hunk 6]
```diff
@@ -1,12 +1,47 @@
...
+const oauth2Client = new OAuth2(GOOGLE_CLIENT_ID, GOOGLE_CLIENT_SECRET, '/');
+
+const getMailer = () => {
+  oauth2Client.setCredentials({
+    refresh_token: GOOGLE_REFRESH_TOKEN
+  });
+
+  const accessToken = oauth2Client.getAccessToken();
+
+  const transport = nodemailer.createTransport({
```
Comment (adamklepacz) [inline server/mailer/index.js]: just` return nodemailer.createTransport({....`

File: server/mailer/index.js [hunk 7]
```diff
@@ -1,12 +1,47 @@
...
+      type: 'OAuth2',
+      user: GOOGLE_API_USER,
+      clientId: GOOGLE_CLIENT_ID,
+      clientSecret: GOOGLE_CLIENT_SECRET,
+      refreshToken: GOOGLE_REFRESH_TOKEN,
+      accessToken
+    }
+  });
+
+  return transport;
```
Comment (adamklepacz) [inline server/mailer/index.js]: and remove this return

File: server/mailer/index.js [hunk 8]
```diff
@@ -105,7 +140,12 @@ const sendReport = async (host, reportData) => {
     })
   };
 
-  return SendGridMail.send(message);
+  const mailer = getMailer();
```
Comment (adamklepacz) [inline server/mailer/index.js]: remove empty line above

File: server/index.js [hunk 9]
```diff
@@ -32,6 +32,7 @@ mongoose.connect(dbUrl, { useNewUrlParser: true }, unlockLocks);
 mongoose.set('useCreateIndex', true);
 mongoose.set('useFindAndModify', false);
 
+app.enable('trust proxy');
```
Comment (adamklepacz) [inline server/index.js]: extract this string to some enum value 

```
const AppConfigs = Object.freeze({
     PROXY: 'trust proxy'
});
```

and use it like this

`app.enable(AppConfigs.PROXY);`

File: server/mailer/index.js [hunk 10]
```diff
@@ -1,12 +1,44 @@
...
+  GOOGLE_CLIENT_ID,
+  GOOGLE_CLIENT_SECRET,
+  GOOGLE_REFRESH_TOKEN,
+  SENDGRID_API_KEY
+} = process.env;
+const {
+  auth: { OAuth2 }
+} = google;
+const oauth2Client = new OAuth2(GOOGLE_CLIENT_ID, GOOGLE_CLIENT_SECRET, '/');
+
```
Comment (adamklepacz) [inline server/mailer/index.js]: rremove empty line

File: server/common/variables.js [hunk 11]
```diff
@@ -89,8 +89,13 @@ const EmailReportsFrequency = enumerable('report')(
   'WEEKLY'
 );
 
+const AppConfig = Object.freeze({
+  PROXY: 'trust proxy'
+});
```
Comment (adamklepacz) [inline server/common/variables.js]: i would place this unders
server/config/enums.js

**Pull request merged**